example/bufferCopy/src/bufferCopy.cpp

```diff
@@ -257,3 +246,19 @@ auto main() -> int
 
     return EXIT_SUCCESS;
 }
+
+auto main() -> int
+{
+    // Execute the example once for each enabled accelerator.
+    // If you would like to execute it for a single accelerator only you can use the following code.
+    //  \code{.cpp}
+    //  auto tag = TagCpuSerial;
+    //  return example(tag);
+    //  \endcode
+    //
+    // valid tags:
+    //   TagCpuSerial, TagGpuHipRt, TagGpuCudaRt, TagCpuOmp2Blocks, TagCpuTbbBlocks,
+    //   TagCpuOmp2Threads, TagCpuSycl, TagCpuTbbBlocks, TagCpuThreads,
+    //   TagFpgaSyclIntel, TagGenericSycl, TagGpuSyclIntel
+    return alpaka::executeForEachAccTag([=](auto const& tag) { return example(tag); });
```

Comment:
To honest, I'm not sure if we should use tags directly. Now it looks like, that tags are required to develop an alpaka application.

Maybe we should use the tags only internal in the `alpaka::executeForEachAccTag` function, which should become ``alpaka::executeForEachAcc`.

Reply:
I do not see a way to implement it as easy as it is now. The reason is that we can not create an instance of Acc, so we can not pass it to the execution lambda.
I also prefer that we advertise tags more because of the reason that you are not allowed to instantiate tags on the host side. This avoids using accelerators as an argument in a lambda.
Additionally accelerators have the problem that the definition are not available if the backend is not activated in CMake.

Reply:
There is another advantage of using tags. Tags only say to which accelerator you like to map you problem but is not say anything about the dimensionality of the index domain. This provides the freedom to switch dimensionality for each kernel depending on the needs.

Reply:
IMO

```C++
return example(tag);
```
vs.
```C++
return example<alpaka::TagToAcc<decltype(TagToAcc)>>();
```

The problem is any way that the dimension and index type for the accelerator is required to create an accelerator.
So it will be 

```C++
return example<alpaka::TagToAcc<decltype(TagToAcc), Dim, Idx>();
```

Where Dim and Idx must moved away from the problem into the main function.